### PR TITLE
React Sample App

### PR DESCRIPTION
**Pull request by velmurugan-balasubramanian**

### CODE REVIEW

File: Demo-Apps/React_sample/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+## React Todo app for freshdesk
+
+This is a sample react app for freshdesk. 
+
+
+### About the app 
+
+the app uses react framework through CDN to demonstrate the use of react in a freshdesk app. the app is a simple todo that uses data storage to save and retriev note
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/README.md]: First letter capital for `the`.
Typo on `retrieve`.

File: Demo-Apps/React_sample/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+### how to run 
+
+1. Clone the repo and navigate to this foler
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/README.md]: Typo for `folder`.

File: Demo-Apps/React_sample/app/app.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            <div key={index}>
+              <input checked={todo.state} type="checkbox" id={todo.index} onChange={() => this.handleChecked(index)} name={todo.index} value={todo.state} />
+              <label for={todo.index}>{todo.todo}</label>
+            </div>
+          )
+        }
+
+      </div>
+    );
+  }a
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/app/app.js]: Mistyped `a` in here.

File: Demo-Apps/React_sample/app/app.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  }
+
+  /**
+   * Update the status of todo 
+   * @param {Object} todo current todo object
+   */
+  updateTodos(todo) {
+    client.db.update("todos", "append", { "todos": [todo] }).then(
+      function (data) {
+        console.log('data Updated', data)
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/app/app.js]: Can we add more context to the error message? The data is a very generic word.

Also, can we add a showNotify here to notify the user?

File: Demo-Apps/React_sample/app/app.js [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+      },
+      function (error) {
+        console.error('error', error)
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/app/app.js]: Please add a detailed error message.

Also, can we add a showNotify here as well?

File: Demo-Apps/React_sample/config/iparam_test_data.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (ravirajthedeveloper) [inline Demo-Apps/React_sample/config/iparam_test_data.json]: Since local testing with this file is deprecated, can you please remove this file?

File: Demo-Apps/React_sample/app/app.js (hunk 5 above)
Comment (hemchander23) [reply Demo-Apps/React_sample/app/app.js]: Yes, please. Notification is important

File: Demo-Apps/React_sample/app/app.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+          </div>
+            , document.getElementById('mydiv'))
+        });
+    });
+});
+
+/**
+ * React Component to list todo's 
+ */
+class Todolist extends React.Component {
```
Comment (hemchander23) [inline Demo-Apps/React_sample/app/app.js]: Can we isolate this react component into a separate JS file?

File: Demo-Apps/React_sample/app/app.js [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+      </div>
+    );
+  }
+}
+
+/**
+ * React Component to add Todos
+ */
+class Todo extends React.Component {
```
Comment (hemchander23) [inline Demo-Apps/React_sample/app/app.js]: Can you isolate this component into separate js? The structure would be hard to follow and is not scalable with more components

File: Demo-Apps/React_sample/app/app.js [hunk 9]
```diff
@@ -0,0 +1,138 @@
+$(document).ready(function () {
+  app.initialized()
+    .then(function (_client) {
+      window.client = _client;
+      client.events.on('app.activated',
+        function () {
+          ReactDOM.render(<div>
+            <Todolist />
```
Comment (hemchander23) [inline Demo-Apps/React_sample/app/app.js]: Why is `<Todo />` out of `<Todolist />` 🤔 Shouldn't this be inside?

```
<Todolist>
    <Todo ..... />
    .
    .
</Todolist>

File: Demo-Apps/React_sample/README.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
+## React Todo app for freshdesk
+
+This is a sample react app for freshdesk.
```
Comment (hemchander23) [inline Demo-Apps/React_sample/README.md]: `React` and `Freshdesk` to contain a capital first letter.

File: Demo-Apps/React_sample/README.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+## React Todo app for freshdesk
+
+This is a sample react app for freshdesk. 
+
+
+### About the app 
+
+The app uses react framework through CDN to demonstrate the use of react in a freshdesk app. the app is a simple todo that uses data storage to save and retrieve note
```
Comment (hemchander23) [inline Demo-Apps/React_sample/README.md]: React Library, not framework I guess. [Refer here](https://www.oreilly.com/library/view/what-react-is/9781491996744/ch01.html)

Comment (hemchander23) [reply Demo-Apps/React_sample/README.md]: Also, take care of the casing.

**Comment by Saif-Shines** [comment]
Can you please change the name of the folder from `React_sample` to `react_todo_app` ?

File: Demo-Apps/React_sample/README.md (hunk 11 above)
Comment (hemchander23) [reply Demo-Apps/React_sample/README.md]: Quoting the article:

`...React Is a Library, Not a Framework
One important distinction between libraries like React and frameworks like Ember.js and AngularJS is that React is concerned only with rendering the UI and leaves many things up to each project to put together....`

File: Demo-Apps/React_sample/app/app.js (hunk 9 above)
Comment (velmurugan-balasubramanian) [reply Demo-Apps/React_sample/app/app.js]: Components are independent, they just share data through the state! so they are at the same level!
